keras/backend/tensorflow_backend.py

```diff
@@ -432,6 +436,10 @@ def switch(condition, then_expression, else_expression):
     return tf.python.control_flow_ops.cond(condition,
                                            lambda: then_expression,
                                            lambda: else_expression)
+def broadcasting_select(condition, A, B):
```

Comment:
Yes it's definitely not intended to be an exposed backend function. I can scope it to the `rnn()` function. I suppose I could remove it altogether, since I only do it in 2 places. I found this perhaps more readable because otherwise the `tile` call is pretty inscrutable. 

The idea is, that `tf.select` needs its condition tensor to be the same shape as A and B, but in our case the condition (mask) tensor is `(nsamples, 1)`, and A and B are `(nsamples, ndimensions)`. So I want to broadcast the mask to match the shape of A and B. That's what the `tile` call does, is just repeat the mask along its second dimension `ndimensions` times.